projects/postgresql/fuzzer/fuzzer_initialize.c

```diff
@@ -35,35 +35,44 @@
 #include "utils/snapmgr.h"
 #include "utils/timeout.h"
 
+#include <libgen.h>
+
 const char *progname;
 static MemoryContext row_description_context = NULL;
 static StringInfoData row_description_buf;
 static const char *username = "username";
 
-int FuzzerInitialize(char *dbname){
-  char *argv[5];
+int FuzzerInitialize(char *dbname, char ***argv){
+  char *av[5];
   char arg_path[50];
   char path_to_db[50];
   char untar[100];
+  char *exe_path = (*argv)[0];
+  //dirname() can modify its argument
+  char *exe_path_copy = strdup(exe_path);
+  char *dir = dirname(exe_path_copy);
+  chdir(dir);
+  free(exe_path_copy);
+
   snprintf(arg_path, sizeof(arg_path), "/tmp/%s/data", dbname);
   snprintf(path_to_db, sizeof(path_to_db), "-D\"/tmp/%s/data\"", dbname);
-  snprintf(untar, sizeof(untar), "rm -rf /tmp/%s && mkdir /tmp/%s && tar -xvf data.tar.gz -C /tmp/%s", dbname, dbname, dbname);
-  
-  argv[0] = "tmp_install/usr/local/pgsql/bin/postgres";
-  argv[1] = path_to_db;
-  argv[2] = "-F";
-  argv[3] = "-k\"/tmp/pg_dbfuzz\"";
-  argv[4] = NULL;
+  snprintf(untar, sizeof(untar), "rm -rf /tmp/%s; mkdir /tmp/%s; tar -xvf data.tar.gz -C /tmp/%s", dbname, dbname, dbname);
```

Comment:
This likely won't work for most engines as tar isn't available in the minijail sandbox. This also seems very brittle. Can the build be modified to have these files extracted already, rather than requiring the target to extract on initialisation?